tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/errors/BigQuerySqlError.java

```diff
@@ -0,0 +1,21 @@
+package com.google.cloud.bigquery.utils.queryfixer.errors;
+
+import com.google.cloud.bigquery.BigQueryException;
+import com.google.cloud.bigquery.utils.queryfixer.entity.Position;
+
+import lombok.Getter;
+
+/** The base class to represent the SQL Error generated by the BigQuery server. It should be resolved from
+ * {@link BigQueryException}. */
+@Getter
+public abstract class BigQuerySqlError {
+
+  BigQueryException errorSource;
+  /** The occurrence of the error at a query. It may be null if the error does not belong to any specific locations. */
```

Comment:
Should they be public fields?

Reply:
I think they should be `protected`, because they will be accessed by the subclasses.

Reply:
Hmm, after I rechecked the code, `private` should also work, I will change to `private`.